pubsublite/streaming-analytics/src/test/java/PubsubliteToGcsIT.java

```diff
@@ -13,79 +13,209 @@
 import org.apache.beam.sdk.testing.TestPipeline;
-import org.apache.beam.sdk.testing.TestStream;
-import org.apache.beam.sdk.transforms.windowing.AfterProcessingTime;
-import org.apache.beam.sdk.transforms.windowing.FixedWindows;
-import org.apache.beam.sdk.transforms.windowing.IntervalWindow;
-import org.apache.beam.sdk.transforms.windowing.Repeatedly;
-import org.apache.beam.sdk.transforms.windowing.Window;
-import org.apache.beam.sdk.values.PCollection;
-import org.apache.beam.sdk.values.TimestampedValue;
-import org.joda.time.Duration;
-import org.joda.time.Instant;
+import org.apache.beam.sdk.values.KV;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 
 public class PubsubliteToGcsIT {
   @Rule public final TestPipeline testPipeline = TestPipeline.create();
 
-  private Instant baseTime = new Instant(0);
+  private static final String projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private String cloudRegion = "us-east1";
+  private final char zoneId = 'b';
+  private static final String suffix = UUID.randomUUID().toString();
+  private static final String topicId = "pubsublite-streaming-analytics";
+  private static final String subscriptionId = "pubsublite-streaming-analytics-" + suffix;
+  private static final String bucketName = "pubsublite-it";
+  private static final String directoryPrefix = "samples/" + suffix;
+  private static final String jobName = "pubsublite-dataflow-job-" + suffix;
+
+  private final Storage storage =
+      StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+
+  private final TopicPath topicPath =
+      TopicPath.newBuilder()
+          .setProject(ProjectId.of(projectId))
+          .setLocation(CloudZone.of(CloudRegion.of(cloudRegion), zoneId))
+          .setName(TopicName.of(topicId))
+          .build();
+
+  private final SubscriptionPath subscriptionPath =
+      SubscriptionPath.newBuilder()
+          .setLocation(CloudZone.of(CloudRegion.of(cloudRegion), zoneId))
+          .setProject(ProjectId.of(projectId))
+          .setName(SubscriptionName.of(subscriptionId))
+          .build();
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    //  Create a test subscription.
+    Subscription subscription =
+        Subscription.newBuilder()
+            .setDeliveryConfig(
+                DeliveryConfig.newBuilder()
+                    .setDeliveryRequirement(DeliveryRequirement.DELIVER_IMMEDIATELY))
+            .setName(subscriptionPath.toString())
+            .setTopic(topicPath.toString())
+            .build();
+
+    AdminClientSettings adminClientSettings =
```

Comment:
Does this need to be created before each test and created to tear down later?

Reply:
Good point, no, not really. I can move them out of the `@Before` and `@After`.